### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by machineglow** [ztalbot2000/homebridge-cmd4#111]
**Describe The Bug:**
homebridge dies with v4.2.0

**To Reproduce:**
on startup, CMD4 throws an error and causes a homebridge bootloop

**Expected Behaviour:**
expect homebridge to keep running

**Paste of Logs:**
```
Jul 24 18:18:55  homebridge[108748]: [7/24/2021, 6:18:55 PM] TypeError: qCallback is not a function
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/homebridge-cmd4/Cmd4Accessory.js:997:27)
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.emit (events.js:327:22)
Jul 24 18:18:55 homebridge[108748]:     at maybeClose (internal/child_process.js:1048:16)
Jul 24 18:18:55 homebridge[108748]:     at Socket.<anonymous> (internal/child_process.js:439:11)
Jul 24 18:18:55 homebridge[108748]:     at Socket.emit (events.js:315:20)
Jul 24 18:18:55 homebridge[108748]:     at Pipe.<anonymous> (net.js:673:12)
Jul 24 18:18:55 homebridge[108748]: [7/24/2021, 6:18:55 PM] TypeError: qCallback is not a function
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/homebridge-cmd4/Cmd4Accessory.js:997:27)
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.emit (events.js:327:22)
Jul 24 18:18:55 homebridge[108748]:     at maybeClose (internal/child_process.js:1048:16)
Jul 24 18:18:55 homebridge[108748]:     at Socket.<anonymous> (internal/child_process.js:439:11)
Jul 24 18:18:55 homebridge[108748]:     at Socket.emit (events.js:315:20)
Jul 24 18:18:55 homebridge[108748]:     at Pipe.<anonymous> (net.js:673:12)
Jul 24 18:18:55 homebridge[108748]: [7/24/2021, 6:18:55 PM] TypeError: qCallback is not a function
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/homebridge-cmd4/Cmd4Accessory.js:997:27)
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.emit (events.js:327:22)
Jul 24 18:18:55 homebridge[108748]:     at maybeClose (internal/child_process.js:1048:16)
Jul 24 18:18:55 homebridge[108748]:     at Socket.<anonymous> (internal/child_process.js:439:11)
Jul 24 18:18:55 homebridge[108748]:     at Socket.emit (events.js:315:20)
Jul 24 18:18:55 homebridge[108748]:     at Pipe.<anonymous> (net.js:673:12)
Jul 24 18:18:55 homebridge[108748]: [7/24/2021, 6:18:55 PM] Got SIGTERM, shutting down Homebridge...
Jul 24 18:18:55 homebridge[108748]: [7/24/2021, 6:18:55 PM] [Harmony] INFO - shutdown
Jul 24 18:18:55 homebridge[108748]: [7/24/2021, 6:18:55 PM] [Aedes MQTT] Stopping Aedes MQTT broker "Aedes MQTT".
Jul 24 18:18:55 homebridge[108748]: [7/24/2021, 6:18:55 PM] TypeError: qCallback is not a function
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/homebridge-cmd4/Cmd4Accessory.js:997:27)
Jul 24 18:18:55 homebridge[108748]:     at ChildProcess.emit (events.js:327:22)
Jul 24 18:18:55 homebridge[108748]:     at maybeClose (internal/child_process.js:1048:16)
Jul 24 18:18:55 homebridge[108748]:     at Socket.<anonymous> (internal/child_process.js:439:11)
Jul 24 18:18:55 homebridge[108748]:     at Socket.emit (events.js:315:20)
Jul 24 18:18:55 homebridge[108748]:     at Pipe.<anonymous> (net.js:673:12)

```

**Cmd4 Config:**

```json
        {
            "platform": "Cmd4",
            "name": "Cmd4",
            "allowTLV8": false,
            "outputConstants": false,
            "restartRecover": true,
            "fetch": "Polled",
            "accessories": [
                {
                    "type": "Switch",
                    "displayName": "Playroom AirPlaying",
                    "on": "FALSE",
                    "name": "Playroom AirPlaying",
                    "Manufacturer": "Apple Inc.",
                    "Model": "AirPort10,115",
                    "SerialNumber": "C86NQKFFFFFF",
                    "polling": true,
                    "interval": 10,
                    "timeout": 10000,
                    "state_cmd_prefix": "bash",
                    "state_cmd": "/var/homebridge/Cmd4Scripts/LivingRoomAirplayState.sh"
                },
                {
                    "type": "Switch",
                    "displayName": "Master Bedroom AirPlaying",
                    "on": "FALSE",
                    "name": "Master Bedroom AirPlaying",
                    "Manufacturer": "Apple Inc.",
                    "Model": "AirPort10,115",
                    "SerialNumber": "C86NQKFFFFFF",
                    "polling": true,
                    "interval": 10,
                    "timeout": 10000,
                    "state_cmd_prefix": "bash",
                    "state_cmd": "/var/homebridge/Cmd4Scripts/MasterBedroomAirplayState.sh",
                    "state_cmd_suffix": ""
                },
                {
                    "type": "Switch",
                    "displayName": "Kitchen AirPlaying",
                    "on": "FALSE",
                    "name": "Kitchen AirPlaying",
                    "Manufacturer": "Apple Inc.",
                    "Model": "AirPort10,115",
                    "SerialNumber": "C86NQKFFFFFF",
                    "polling": true,
                    "interval": 10,
                    "timeout": 10000,
                    "state_cmd_prefix": "bash",
                    "state_cmd": "/var/homebridge/Cmd4Scripts/KitchenAirplayState.sh",
                    "state_cmd_suffix": ""
                },
                {
                    "type": "Switch",
                    "displayName": "Family Room AirPlaying",
                    "on": "FALSE",
                    "name": "Family Room AirPlaying",
                    "Manufacturer": "Apple Inc.",
                    "Model": "AirPort10,115",
                    "SerialNumber": "C86NQKFFFFFF",
                    "polling": true,
                    "interval": 20,
                    "timeout": 10000,
                    "state_cmd_prefix": "bash",
                    "state_cmd": "/var/homebridge/Cmd4Scripts/FamilyRoomAirplayState.sh",
                    "state_cmd_suffix": ""
                }
            ]
        },


```

**Environment:**

* **Node.js Version**: 14.16
* **NPM Version**: 7.6.1
* **Homebridge Version**: 1.3.4
* **homebridge-cmd4 Version**: 4.2.0
* **Operating System**: Ubuntu 
* **Process Supervisor**: Systemd

**Comment by ztalbot2000** [comment]
Hi,

  Nasty.  I'll look into it.

John

**Comment by ztalbot2000** [comment]
Hi,

  Weird,  I essentially have the same environment.  Can you please do me a
favor and on the offending line change it to:

If (qCallback != null)

Thanks,
John

**Comment by machineglow** [comment]
Sure. I reverted back to 4.0.1 for now but will switch back and do the code change later tonight. 

> Hi,
> 
>   Weird,  I essentially have the same environment.  Can you please do me a
> favor and on the offending line change it to:
> 
> If (qCallback != null)
> 
> Thanks,
> John
>

**Comment by ztalbot2000** [comment]
I was able to replicate it using your config.  I will fix it ASAP. Though I
do not know why it fails yet.

Ttyl,
John

**Comment by ztalbot2000** [comment]
Hi,

   Not that I like the way I had to fix this, but it will work in a pinch.
Resolved in 4.2.1.  

Sorry for the inconvenience.  I will look at why my Unit tests did not pick this up.

Take care,
John Talbot
